source/mir/ndslice/topology.d

```diff
@@ -3067,3 +3068,784 @@ unittest
     assert(lowerData == [0, 3, 4, 6, 7, 8]);
     assert(upperData == [0, 1, 2, 4, 5, 8]);
 }
+
+/// Transposed adjusted to ignore dim=0 and include universal
+@fastmath private template adjTransposed(Dimensions...)
+    if (Dimensions.length)
+{
+    import mir.ndslice.slice : Slice, SliceKind;
+    import mir.ndslice.internal : isSize_t, toSize_t;
+    import std.meta : allSatisfy, staticMap;
+   
+    static if (!allSatisfy!(isSize_t, Dimensions))
+        alias adjTransposed = .adjTransposed!(staticMap!(toSize_t, Dimensions));
+    else
+    ///
+    auto adjTransposed(SliceKind kind, size_t[] packs, Iterator)(Slice!(kind, packs, Iterator) slice)
+    {
+        import mir.ndslice.topology : ipack;
+        import mir.ndslice.internal : DimensionsCountCTError, DimensionCTError;
+       
+        mixin DimensionsCountCTError;
+            
+        static if (Dimensions.length == 1 && Dimensions[0] == 0)
+        {
+            return slice;
+        }
+        else
```

Comment:
Please add specialization for `canonical` case. Canonical case is valid when the last (last of unpacked slice) dimension of contiguous slice is not moved upper. The code example for single dimension is below. Please generalize it for multiple dimensions.

```d
...
else static if (dim + 1 < packs[0] || packs.length > 1)
{
            import mir.ndslice : canonical, transposed;
            return slice.canonical.transposed!dim;
} 
...
```
please cover this in unittest (contiguous -> canonical) . Existing unittests should be enough, just extend them